### DISCUSSION TITLE
about tauri-plugin-global-shortcut and tauri-plugin-autostart

### DISCUSSION THREAD

**Discussion by ggslayer** [tauri-apps/tauri#8543] [Q&A] [answered]
I'm using v2.0 alpha version. And I want to add shortcut and autostart feature to my app.

according to: https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/global-shortcut
"tauri_plugin_shortcut::init()"  can't though rust compile, because the plugin struct is "tauri_plugin_global_shortcut". but tauri_plugin_global_shortcut has no init method too.    so, I don't know how to use it in rust code.

The shortcut plugin, is there some demo code to call plugin's "enable" method in rust?  

Thank you for your help

**Comment by FabianLars** [comment]
For the global-shortcut plugin, checkout our example app we use for testing: [plugin init](https://github.com/tauri-apps/plugins-workspace/blob/v2/examples/api/src-tauri/src/lib.rs#L45) & [shortcut registration in js (svelte)](https://github.com/tauri-apps/plugins-workspace/blob/v2/examples/api/src/views/Shortcuts.svelte) though the rust docs in the readme should be still correct for that.

**Reply by ggslayer** [reply]
thank you for your reply,  my question missing some info..  

1. is there some "rust" demo code to register shortcut  and not in js?
2. is there some "rust" demo code to plugin autostart usage?

**Reply by FabianLars** [reply, marked as answer]
Not yet, no, but here's the gist of it:
```rs
use tauri_plugin_autostart::ManagerExt;
use tauri_plugin_global_shortcut::{Code, GlobalShortcutExt, Modifiers};

fn main() {
    tauri::Builder::default()
        .plugin(tauri_plugin_autostart::init(
            tauri_plugin_autostart::MacosLauncher::LaunchAgent,
            None,
        ))
        .plugin(
            tauri_plugin_global_shortcut::Builder::with_handler(|app_handle, shortcut| {
                // Low effor solution. I'm sure there's a nicer way.
                if shortcut.matches(Modifiers::CONTROL, Code::KeyE) {
                    // do something
                }
            })
            .build(),
        )
        .setup(|app| {
            app.handle()
                .autolaunch()
                .enable()
                .expect("app should enable autostart");

            let global_shortcut = app.handle().global_shortcut();
            global_shortcut
                .register("CmdOrControl+E")
                .expect("app should register cmd+E hotkey");

            Ok(())
        })
        .run(tauri::generate_context!())
        .expect("error while running tauri application")
```

**Reply by ggslayer** [reply]
Got it!  thanks!
